source.go

```diff
@@ -283,19 +284,19 @@ func WithCreateDestPath() llb.CopyOption {
 	})
 }
 
-func SourceIsDir(src Source) (bool, error) {
+func SourceIsDir(src Source) bool {
 	switch {
 	case src.DockerImage != nil,
 		src.Git != nil,
 		src.Build != nil,
 		src.Context != nil:
-		return true, nil
+		return true
 	case src.HTTP != nil:
-		return false, nil
+		return false
 	case src.Inline != nil:
-		return src.Inline.Dir != nil, nil
+		return src.Inline.Dir != nil
 	default:
-		return false, fmt.Errorf("unsupported source type")
+		panic("unreachable")
```

Comment:
In the unlikely event that it did reach here, would it be a good idea to have a verbose error message?
```suggestion
		panic("unreachable: unsupported source type")
```

Reply:
Unless I am mistaken, I don't think the panic message gets propagated back to the client if the frontend process panics. @cpuguy83 is this right?

Reply:
Correct, its only in the daemon logs.
Though we could possibly recover panics and send them back as errors.

Reply:
I'm curious why remove the error?
I guess at this point it is a bug, but maybe, worth keeping the error and calling out it is a bug that should be filed.

Reply:
Having just hit this with the jammy PR, it is indeed really annoying to have this error.
I think its fine to panic.

Reply:
> Correct, its only in the daemon logs. Though we could possibly recover panics and send them back as errors.

I like this idea, because I've run into unexplained panics before and it's been a bit of a pain to troubleshoot.,